File: TransCoder_tokenized_test_set_functions/python/SPACE_OPTIMIZED_DP_SOLUTION_0_1_KNAPSACK_PROBLEM_1.py
def KnapSack ( val, wt, n, W ) :
    dp = [ 0 ] * ( W + 1 );
    
    for i in range ( n ) :
        for j in range ( W, wt [ i ], - 1 ) :
            dp [ j ] = max ( dp [ j ], val [ i ] + dp [ j - wt [ i ] ] );
            
    return dp [ W ];

File: TransCoder_tokenized_test_set_functions/java/SPACE_OPTIMIZED_DP_SOLUTION_0_1_KNAPSACK_PROBLEM_1.java
static int KnapSack ( int val [ ], int wt [ ], int n, int W ) {
  int [ ] dp = new int [ W + 1 ];
  Arrays . fill ( dp, 0 );
  for ( int i = 0;
  i < n;
  i ++ ) for ( int j = W;
  j >= wt [ i ];
  j -- ) dp [ j ] = Math . max ( dp [ j ], val [ i ] + dp [ j - wt [ i ] ] );
  return dp [ W ];
}